middle_end/flambda2/to_cmm/to_cmm.ml

```diff
@@ -114,31 +119,24 @@ let unit0 ~offsets flambda_unit ~all_code =
     C.create_ccatch ~rec_flag:false ~body
       ~handlers:[C.handler ~dbg return_cont return_cont_params unit_value]
   in
+  let entry_name = Cmm_helpers.make_symbol "entry" in
+  let res, entry_sym = R.raw_symbol res ~global:Global entry_name in
   let entry =
     (* CR mshinwell: This should at least be given a source file location. *)
     let dbg = Debuginfo.none in
-    let fun_name = Cmm_helpers.make_symbol "entry" in
     let fun_codegen =
       let fun_codegen = [Cmm.Reduce_code_size; Cmm.Use_linscan_regalloc] in
       if Flambda_features.backend_cse_at_toplevel ()
       then fun_codegen
       else Cmm.No_CSE :: fun_codegen
     in
-    C.cfunction
-      (C.fundecl
-         (Cmm.global_symbol fun_name)
-         [] body fun_codegen dbg Default_poll)
+    C.cfunction (C.fundecl entry_sym [] body fun_codegen dbg Default_poll)
   in
   let { R.data_items; gc_roots; functions } = R.to_cmm res in
-  let cmm_helpers_data = flush_cmm_helpers_state () in
-  let gc_root_data =
-    C.gc_root_table
-      (List.map
-         (fun sym -> Linkage_name.to_string (Symbol.linkage_name sym))
-         gc_roots)
-  in
+  let cmm_helpers_data = flush_cmm_helpers_state res in
```

Comment:
I'd prefer to ignore the `res` here instead of not returning it, just in case we ever add some other functions using the `res` afterwards.